inst/extdata/mvp/product-carbon-transition-risk_review.Rmd

```diff
@@ -0,0 +1,308 @@
+  select(all_of(c(
+    "Activity Name",
+    "Geography",
+    "Time Period",
+    "Special Activity Type",
+    "Sector",
+    "ISIC Classification",
+    "ISIC Section",
+    "Reference Product Name",
+    "CPC Classification",
+    "Unit"
+  )))
+
+# renaming
+names(data_short)[names(data_short) == "Activity Name"] <- "activity"
+names(data_short)[names(data_short) == "Geography"] <- "geo"
+names(data_short)[names(data_short) == "Time Period"] <- "time"
+names(data_short)[names(data_short) == "Special Activity Type"] <- "special_act_type"
+names(data_short)[names(data_short) == "Sector"] <- "sec"
+names(data_short)[names(data_short) == "ISIC Classification"] <- "isic_class"
+names(data_short)[names(data_short) == "ISIC Section"] <- "isic_sec"
+names(data_short)[names(data_short) == "Reference Product Name"] <- "ref_prod"
+names(data_short)[names(data_short) == "CPC Classification"] <- "cpc_class"
+names(data_short)[names(data_short) == "Unit"] <- "unit"
+```
+
+## 3. Add column “production_emissions_per_product”
+
+``` {r}
+# create column with sample emission data, using "rnorm" with normal distribution
+values <- rnorm(nrow(data_short), mean = 6, sd = 1.5)
+data_short <- data_short %>%
+  mutate(production_emissions = values, emissions_unit = "per product unit")
+```
+
+## B) Calculation of transition risks
+
+## 1. Calculate percentiles in comparison to different sub-sets
+
+``` {r}
+# percentile of all products: 'perc_all'
+data_short <- data_short %>%
+  mutate(perc_all = rank(production_emissions) / length(production_emissions))
+
+# percentile of all products with same unit: 'perc_unit'
+data_short <- data_short %>%
+  group_by(unit) %>%
+  mutate(perc_unit = rank(production_emissions) / length(production_emissions)) %>%
+  ungroup()
+
+# percentile of all products with same sector 'ISIC Classification' and 'Unit':
+# 'perc_unit_isic'
+data_short <- data_short %>%
+  group_by(unit, isic_class) %>%
+  mutate(perc_unit_isic = rank(production_emissions) / length(production_emissions)) %>%
+  ungroup()
+```
+
+## 2. Assign scores to the activities based on their position within the distribution
+
+``` {r}
+# assign scores to percentiles
+data_short <- data_short %>%
+  mutate(score_all = case_when(
+    perc_all < 0.3 ~ "low",
+    perc_all >= 0.3 & perc_all < 0.7 ~ "medium",
+    perc_all >= 0.7 ~ "high",
+  ))
+
+data_short <- data_short %>%
+  mutate(score_unit = case_when(
+    perc_unit < 0.33 ~ "low",
+    perc_unit >= 0.3 & perc_unit < 0.7 ~ "medium",
+    perc_unit >= 0.7 ~ "high",
+  ))
+
+data_short <- data_short %>%
+  mutate(score_unit_isic = case_when(
+    perc_unit_isic < 0.3 ~ "low",
+    perc_unit_isic >= 0.3 & perc_unit_isic < 0.7 ~ "medium",
+    perc_unit_isic >= 0.7 ~ "high",
+  ))
+
+# show distributions (each score should have same probability for score_all)
+barplot(prop.table(table(data_short$score_all)))
+barplot(prop.table(table(data_short$score_unit)))
+barplot(prop.table(table(data_short$score_unit_isic)))
```

Comment:
This suggests we need a additional function that creates a plot. A function should either compute something (e.g. `company_scores`) or have a side effect (e.g. plot or save a file) but not both.

Here is what the code might eventually look like in ggplot2 (but don't worry about it now because you want to deliver just a .csv file with company scores).

![image](https://user-images.githubusercontent.com/5856545/227338862-a7bad893-0f2f-4651-875a-edf46ec6c5f2.png)